src/metabase/lib/join.cljc

```diff
@@ -464,6 +464,16 @@
     (run! generator taken-names)
     (generator base-name)))
 
+(defn default-alias
+  "Generate a default `:alias` for a join clause"
+  [query stage-number a-join]
+  (let [stage       (lib.util/query-stage query stage-number)
+        home-cols   (lib.metadata.calculation/visible-columns query stage-number stage)
```

Comment:
This is duplicating the `visible-columns` call with `add-default-alias` below. (It's not cached in the 3-arity form.)